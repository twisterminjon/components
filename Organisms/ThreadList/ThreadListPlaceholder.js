import React, { Component } from 'react'
import PropTypes from 'prop-types'

import ThreadListCardPlaceholder from './ThreadListCardPlaceholder'

import './ThreadList.css'

export default class ThreadListPlaceholder extends Component {
  static propTypes = {
    /** ClassName for the wrapper */
    className: PropTypes.string,
    /** Style for the wrapper */
    style: PropTypes.object,
  }
  static defaultProps = {
    className: '',
    style: {},
  }

  render() {
    const { className, style } = this.props

    let placeholderRender = []
    for (let i = 0; i < 10; i++) {
      let time = i * 0.5

      placeholderRender.push(<ThreadListCardPlaceholder delay={time} className={className} key={time} />)
    }

    return (
      <div className={`messagelist-list ${className}`} style={style}>
        {placeholderRender}
      </div>
    )
  }
}
