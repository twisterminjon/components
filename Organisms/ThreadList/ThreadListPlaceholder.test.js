import React from 'react'
import ThreadListPlaceholder from './ThreadListPlaceholder'

describe('ThreadListPlaceholder', () => {
  it('matches the snapshot', () => {
    const wrapper = window.shallow(<ThreadListPlaceholder />)
    expect(wrapper).toMatchSnapshot()
  })
})
