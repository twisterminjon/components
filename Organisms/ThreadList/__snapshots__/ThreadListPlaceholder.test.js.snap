// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadListPlaceholder matches the snapshot 1`] = `
<div
  className="messagelist-list "
  style={Object {}}
>
  <ThreadListCardPlaceholder
    className=""
    delay={0}
    key="0"
    style={Object {}}
  />
  <ThreadListCardPlaceholder
    className=""
    delay={0.5}
    key="0.5"
    style={Object {}}
  />
  <ThreadListCardPlaceholder
    className=""
    delay={1}
    key="1"
    style={Object {}}
  />
  <ThreadListCardPlaceholder
    className=""
    delay={1.5}
    key="1.5"
    style={Object {}}
  />
  <ThreadListCardPlaceholder
    className=""
    delay={2}
    key="2"
    style={Object {}}
  />
  <ThreadListCardPlaceholder
    className=""
    delay={2.5}
    key="2.5"
    style={Object {}}
  />
  <ThreadListCardPlaceholder
    className=""
    delay={3}
    key="3"
    style={Object {}}
  />
  <ThreadListCardPlaceholder
    className=""
    delay={3.5}
    key="3.5"
    style={Object {}}
  />
  <ThreadListCardPlaceholder
    className=""
    delay={4}
    key="4"
    style={Object {}}
  />
  <ThreadListCardPlaceholder
    className=""
    delay={4.5}
    key="4.5"
    style={Object {}}
  />
</div>
`;
